### PR TITLE
Fix for Issue1925 Point_on_edge_t

### LINKED ISSUE #1925: Making compilation faster  include/c_types/point_on_edge_t.h
Refer to #1893 for instructions and details.

### PR DESCRIPTION
**Pull request by Ri7Sh**
Fixes #1925  .

Changes proposed in this pull request:
- fixing point_on_edge_t.h
- modifications due to the change on point_on_edge_t.h

@pgRouting/admins

### CODE REVIEW

File: src/driving_distance/withPoints_dd_driver.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
  ********************************************************************PGR-GNU*/
 
-
+#include "c_types/point_on_edge_t.h"
```
Comment (cvvergara) [inline src/driving_distance/withPoints_dd_driver.cpp:30]: @Ri7Sh 
Can you update your branch: (where upstream is the main repository)
```
git fetch upstream
git rebase upstream/develop
git push -f
```
There was an adjustment on the style check.
The first included file must be the header of the corresponding file
So  for this file (`withPoints_dd_driver.cpp `) the first include should be:
```
#include "drivers/driving_distance/withPoints_dd_driver.h"
```
and after that goes includes that belong to the system like 
```
#include <sstream>
```
and after those you add the other includes like
```
#include "c_types/point_on_edge_t.h"
```

For this file in particular:
```
#include "drivers/driving_distance/withPoints_dd_driver.h"

#include <sstream>
#include <deque>
#include <vector>
#include <algorithm>

#include "dijkstra/pgr_dijkstra.hpp"
#include "withPoints/pgr_withPoints.hpp"

#include "c_types/point_on_edge_t.h"
#include "cpp_common/pgr_alloc.hpp"
```

File: src/withPoints/pgr_withPoints.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
  ********************************************************************PGR-GNU*/
 
-
+#include "c_types/point_on_edge_t.h"
```
Comment (cvvergara) [inline src/withPoints/pgr_withPoints.cpp:30]: Same problem as above

File: src/withPoints/withPoints_driver.cpp [hunk 3]
```diff
@@ -27,6 +27,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
  ********************************************************************PGR-GNU*/
 
+#include "c_types/point_on_edge_t.h"
```
Comment (cvvergara) [inline src/withPoints/withPoints_driver.cpp:30]: same problem as above

**Review by cvvergara — CHANGES_REQUESTED** [review]

File: src/withPoints/pgr_withPoints.cpp (hunk 2 above)
Comment (Ri7Sh) [reply src/withPoints/pgr_withPoints.cpp]: It fails to build if I move `#include "c_types/point_on_edge_t.h"` below `#include "withPoints/pgr_withPoints.hpp"`

**Comment by cvvergara** [comment]
What ever you did it is building (as shown in the actions)
Only this problem:
https://github.com/pgRouting/pgrouting/pull/2025/checks?check_run_id=3110399454#step:4:160
src/withPoints/pgr_withPoints.cpp

https://github.com/Ri7Sh/pgrouting/blob/issue1925-2/src/withPoints/pgr_withPoints.cpp#L30
Should look like:
```
#include "withPoints/pgr_withPoints.hpp"

#include <sstream>
#include <deque>
#include <set>
#include <vector>
#include <string>
#include <algorithm>
#include <cassert>

#include "c_types/point_on_edge_t.h"
#include "cpp_common/pgr_assert.h"
```
And after you are done, we need maximum 2 commits, 
```
git fetch upstream
git rebase -i upstream/develop
git push -f
```
use fixup


```

**Comment by Ri7Sh** [comment]
@cvvergara I tried the solution, you suggested to change /src/withPoints/pgr_withPoints.cpp from
`#include "c_types/point_on_edge_t.h"
#include "withPoints/pgr_withPoints.hpp"

#include <sstream>
#include <deque>
#include <set>
#include <vector>
#include <string>
#include <algorithm>
#include <cassert>

#include "cpp_common/pgr_assert.h"
`

to 

`#include "withPoints/pgr_withPoints.hpp"

#include <sstream>
#include <deque>
#include <set>
#include <vector>
#include <string>
#include <algorithm>
#include <cassert>

#include "c_types/point_on_edge_t.h"
#include "cpp_common/pgr_assert.h"
`

which fails to build.

**Comment by cvvergara** [comment]
Will have a look

**Comment by cvvergara** [comment]
Do the following:
```
git fetch upstream
git checkout upstream/develop include/withPoints/pgr_withPoints.hpp
git commit -m '[C/C++] adjustments because of change on point_on_edge_t.h'
git push
```
and it will work, I already tested those commands.
I'll mark it as draft while you work on this.
Please click on "Ready for review" when done with this suggestion

**Comment by Ri7Sh** [comment]
@cvvergara Tried the solution as per your advice, I am still getting errors.

**Comment by cvvergara** [comment]
I can see that you are stuck.
I will close this PR and release the issue to anyone.
Better to start over if you want.

**Pull request closed without merging**